From a21f9b66064875d6a6470f881ae55be083fd5b6d Mon Sep 17 00:00:00 2001
From: Venkateshwar Rao G <venkateshwar.rao.gannavarapu@xilinx.com>
Date: Mon, 26 Feb 2018 02:07:04 +0530
Subject: [PATCH 0299/1852] drm: xlnx: scaler: Add YUV420 support

commit e51aa3575f44b411fc9d38f69660ef20613038cc from
https://github.com/Xilinx/linux-xlnx.git

This commit adds the YUV 420 media bus format to the VPSS Scaler Driver

Signed-off-by: Venkateshwar Rao G <venkateshwar.rao.gannavarapu@xilinx.com>
Reviewed-by: Hyun Kwon <hyun.kwon@xilinx.com>
Signed-off-by: Michal Simek <michal.simek@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 drivers/gpu/drm/xlnx/xlnx_scaler.c | 8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/xlnx/xlnx_scaler.c b/drivers/gpu/drm/xlnx/xlnx_scaler.c
index 0cc0493d72b4..8460b1691292 100644
--- a/drivers/gpu/drm/xlnx/xlnx_scaler.c
+++ b/drivers/gpu/drm/xlnx/xlnx_scaler.c
@@ -16,7 +16,6 @@
  * Need to implement in a modular approach to share driver code between
  * V4L2 and DRM frameworks.
  * Should be integrated with plane.
- * Add YUV420 support.
  */
 
 #include <linux/device.h>
@@ -693,6 +692,7 @@ static const u32 xilinx_scaler_video_fmts[] = {
 	MEDIA_BUS_FMT_RGB888_1X24,
 	MEDIA_BUS_FMT_VUY8_1X24,
 	MEDIA_BUS_FMT_UYVY8_1X16,
+	MEDIA_BUS_FMT_VYYUYY8_1X24,
 };
 
 /**
@@ -1250,6 +1250,9 @@ xv_vscaler_setup_video_fmt(struct xilinx_scaler *scaler, u32 code_in)
 	case MEDIA_BUS_FMT_RGB888_1X24:
 		video_in = XVIDC_CSF_RGB;
 		break;
+	case MEDIA_BUS_FMT_VYYUYY8_1X24:
+		video_in = XVIDC_CSF_YCRCB_420;
+		break;
 	default:
 		dev_info(scaler->dev, "Vscaler Unsupported media fmt\n");
 		return -EINVAL;
@@ -1304,6 +1307,9 @@ static int xv_hscaler_setup_video_fmt(struct xilinx_scaler *scaler,
 	case MEDIA_BUS_FMT_RGB888_1X24:
 		video_out = XVIDC_CSF_RGB;
 		break;
+	case MEDIA_BUS_FMT_VYYUYY8_1X24:
+		video_out = XVIDC_CSF_YCRCB_420;
+		break;
 	default:
 		dev_info(scaler->dev, "Hscaler Unsupported Out media fmt\n");
 		return -EINVAL;
-- 
2.31.1

